bugfixtemplates/logging-then-throwing/LoggingThenThrowing.java

```diff
@@ -0,0 +1,63 @@
+package formatint;
+
+import java.io.File;
+
+import com.github.javaparser.utils.Log;
+
+class FormatIntUsingPercentD {
+
+    void log_warning_then_throw(Foo foo) {
+        try {
+            foo.invoke();
+        } catch (Throwable e) {
+            Log.warn(foo.getFailureStatus());
+            throw e;
+        }
+    }
+    
+    void log_error_then_throw(Foo foo) {
+        try {
+            foo.invoke();
+        } catch (FooException e) {
+            Log.error(foo.getFailureStatus());
+            throw e;
+        }
+    }
+    
+    void log_error_then_throw_custom_exception(Foo foo) {
+        try {
+            foo.invoke();
+        } catch (FooException e) {
+            Log.error(foo.getFailureStatus());
+            throw new CustomException(e);
+        }
+    }
+    
+    void log_error_else_throw(Foo foo) {
+        try {
+            foo.invoke();
+        } catch (FooException e) {
+            if (foo.cond()) {
+                Log.error(foo.getFailureStatus());
+            } else {
+                throw e;
+            }
+        }
+    }
+    
+    void just_log(Foo foo) {
+        try {
+            foo.invoke();
+        } catch (FooException e) {
+            Log.error(foo.getFailureStatus());
+        }
+    }
+    
+    void just_throw(Foo foo) {
+        try {
+            foo.invoke();
+        } catch (FooException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)